### PR TITLE
BEI-24: Create the steps table

### PR DESCRIPTION
**Pull request by hizikex**
#### WHY THIS PR?
-We need a database to keep the record of steps to prepare a recipe.

#### WHAT DOES THIS PR?
-It creates table where steps on how to create a recipe are kept.

#### HOW CAN THIS BE TESTED
N/A

#### THINGS YOU MAY WANT THE REVIEWER TO NOTE
N/A

### CODE REVIEW

File: src/core/database/migrations/20230729221602-create-steps.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+'use strict';
+/** @type {import('sequelize-cli').Migration} */
+module.exports = {
+  async up(queryInterface, Sequelize) {
```
Comment (RICHYDOS) [inline src/core/database/migrations/20230729221602-create-steps.js:4]: Why are there 2 create-steps.js migration files?

File: src/core/models/recipe.ts [hunk 2]
```diff
@@ -71,4 +71,9 @@ Recipe.init(
...
   foreignKey: 'user_id',
   as: 'user',
-});
\ No newline at end of file
+});
+
+Recipe.hasMany(Step, {
+  foreignKey: 'recipe_id',
+  as: 'recipe'
+})
```
Comment (RICHYDOS) [inline src/core/models/recipe.ts:79]: Add an extra line here

File: src/core/models/steps.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      recipe_id: {
+        type: DataTypes.INTEGER,
+        allowNull: false,
+        primaryKey: true,
+      },
+      number: {
+        type: DataTypes.INTEGER,
+        allowNull: false,
+        unique: true
+      },
```
Comment (RICHYDOS) [inline src/core/models/steps.ts:29]: I have a question. I get why "number" should be unique but putting unique here means that we can't have a particular number appear twice. For example, for "recipe_1" the number field should have a value of 1 appear only once but putting "unique" means that for "recipe_2" we can't have a number field with a value of 1. Is there another way of setting it here or is this something that we should handle in the controllers?

**Review by RICHYDOS — CHANGES_REQUESTED** [review]

File: src/core/models/steps.ts (hunk 3 above)
Comment (hizikex) [reply src/core/models/steps.ts:29]: That's true actually

File: src/core/interfaces/step.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+export interface StepAttributes {
+    id: number;
+    recipe_id: number;
+    number: number;
+    description: string;
+  }
+
```
Comment (LucianaDera) [inline src/core/interfaces/step.ts:7]: add an extra line

File: src/core/models/recipe.ts [hunk 5]
```diff
@@ -71,4 +71,9 @@ Recipe.init(
 Recipe.belongsTo(User, {
   foreignKey: 'user_id',
   as: 'user',
-});
```
Comment (LucianaDera) [inline src/core/models/recipe.ts:74]: Any reason for an extra line

**Review by LucianaDera — CHANGES_REQUESTED** [review]

**Review by RICHYDOS — APPROVED** [review]

File: src/core/models/recipe.ts [hunk 6]
```diff
@@ -2,7 +2,11 @@ import { Model, DataTypes, CreationOptional, InferCreationAttributes, InferAttri
 import { sequelize } from '../database/sequelize';
 import { RecipeAttributes } from '../interfaces/recipes';
 import { User } from './users';
+<<<<<<< HEAD
+import { Step } from './steps';
+=======
 import { Comment } from './comments';
+>>>>>>> origin
```
Comment (RICHYDOS) [inline src/core/models/recipe.ts:9]: What is happening here?

File: src/core/models/recipe.ts [hunk 7]
```diff
@@ -74,7 +77,14 @@ Recipe.belongsTo(User, {
   as: 'user',
 });
 
+<<<<<<< HEAD
+Recipe.hasMany(Step, {
+  foreignKey: 'recipe_id',
+  as: 'recipe'
+})
+=======
```
Comment (RICHYDOS) [inline src/core/models/recipe.ts:85]: Same question as before... What is happening here

Comment (hizikex) [reply src/core/models/recipe.ts:85]: It is resolved

File: src/core/database/config/database.js [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+  development: {
+    ...databaseConfig
+  },
+  test: {
+    ...databaseConfig
+  },
+  staging: {
+    ...databaseConfig
+  },
+  production: {
```
Comment (RICHYDOS) [inline src/core/database/config/database.js:25]: Got no response regarding this... We already have this file in config.js and sequelize.js. This file is not needed

**Review by RICHYDOS — CHANGES_REQUESTED** [review]

File: src/core/database/config/database.js (hunk 8 above)
Comment (hizikex) [reply src/core/database/config/database.js:25]: It's same file... Not a new one.
It was just renamed.

**Review by RICHYDOS — APPROVED** [review]

**Review by LucianaDera — APPROVED** [review]

**Pull request merged**